cmd/aks-periscope/aks-periscope.go

```diff
@@ -22,25 +23,38 @@ func main() {
 		log.Printf("Failed to create CRD: %+v", err)
 	}
 
+	clusterType := os.Getenv("CLUSTER_TYPE")
+
 	collectors := []interfaces.Collector{}
 	containerLogsCollector := collector.NewContainerLogsCollector(exporter)
-	collectors = append(collectors, containerLogsCollector)
-	systemLogsCollector := collector.NewSystemLogsCollector(exporter)
-	collectors = append(collectors, systemLogsCollector)
 	networkOutboundCollector := collector.NewNetworkOutboundCollector(5, exporter)
-	collectors = append(collectors, networkOutboundCollector)
-	ipTablesCollector := collector.NewIPTablesCollector(exporter)
-	collectors = append(collectors, ipTablesCollector)
-	nodeLogsCollector := collector.NewNodeLogsCollector(exporter)
-	collectors = append(collectors, nodeLogsCollector)
 	dnsCollector := collector.NewDNSCollector(exporter)
-	collectors = append(collectors, dnsCollector)
 	kubeObjectsCollector := collector.NewKubeObjectsCollector(exporter)
-	collectors = append(collectors, kubeObjectsCollector)
+	systemLogsCollector := collector.NewSystemLogsCollector(exporter)
+	ipTablesCollector := collector.NewIPTablesCollector(exporter)
+	nodeLogsCollector := collector.NewNodeLogsCollector(exporter)
 	kubeletCmdCollector := collector.NewKubeletCmdCollector(exporter)
-	collectors = append(collectors, kubeletCmdCollector)
 	systemPerfCollector := collector.NewSystemPerfCollector(exporter)
-	collectors = append(collectors, systemPerfCollector)
+	helmCollector := collector.NewHelmCollector(exporter)
+
+	if strings.EqualFold(clusterType, "connectedCluster") {
+		collectors = append(collectors, containerLogsCollector)
```

Comment:
💡 This all is duplicated now, you could do it this way:

i.e. for conditional if, you have at least 4 duplicate appends. 

Could do something like this: 

```
	collectors = append(collectors, containerLogsCollector)
	collectors = append(collectors, dnsCollector)
	collectors = append(collectors, kubeObjectsCollector)
	collectors = append(collectors, networkOutboundCollector)

	if strings.EqualFold(clusterType, "connectedCluster") {
		collectors = append(collectors, helmCollector)

	} else {
		collectors = append(collectors, systemLogsCollector)
		collectors = append(collectors, ipTablesCollector)
		collectors = append(collectors, nodeLogsCollector)
		collectors = append(collectors, kubeletCmdCollector)
		collectors = append(collectors, systemPerfCollector)
	}
```

Reply:
@Tatsinnit this was originally as you describe above - I suggested changing it to the current form so that all the appends were inside the conditional if / else (not sure where comment has gone).

I think I prefer this over what you are suggesting - otherwise there is a "common" area which has to be modified in the case that we want to add another value for clusterType. 

Don't have strong views either way if @sophsoph321 wants to change it though :)